calyx-py/calyx/builder.py

```diff
@@ -268,53 +277,64 @@ def seq_mem_d1(
 
-    def sub(self, name: str, size: int, signed=False):
+    def add(self, size: int, name=None, signed=False) -> CellBuilder:
+        """Generate a StdAdd cell."""
+        return self.binary_op_helper("add", size, name, signed)
+
+    def sub(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdSub cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("sub", size, signed))
+        return self.binary_op_helper("sub", size, name, signed)
 
-    def gt(self, name: str, size: int, signed=False):
+    def gt(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdGt cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("gt", size, signed))
+        return self.binary_op_helper("gt", size, name, signed)
 
-    def lt(self, name: str, size: int, signed=False):
+    def lt(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdLt cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("lt", size, signed))
+        return self.binary_op_helper("lt", size, name, signed)
 
-    def eq(self, name: str, size: int, signed=False):
+    def eq(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdEq cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("eq", size, signed))
+        return self.binary_op_helper("eq", size, name, signed)
 
-    def neq(self, name: str, size: int, signed=False):
+    def neq(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdNeq cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("neq", size, signed))
+        return self.binary_op_helper("neq", size, name, signed)
 
-    def ge(self, name: str, size: int, signed=False):
+    def ge(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdGe cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("ge", size, signed))
+        return self.binary_op_helper("ge", size, name, signed)
 
-    def le(self, name: str, size: int, signed=False):
+    def le(self, size: int, name=None, signed=False) -> CellBuilder:
         """Generate a StdLe cell."""
-        self.prog.import_("primitives/binary_operators.futil")
-        return self.cell(name, ast.Stdlib.op("le", size, signed))
+        return self.binary_op_helper("le", size, name, signed)
+
+    def logical_op_helper(self, operation, size: int, name=None) -> CellBuilder:
+        """Generate a logical operator cell, of the flavor specified in {operation}."""
+        name = name or f"{operation}_{self.generate_name(operation)}"
+        assert isinstance(name, str)
```

Comment:
I know @rachitnigam made the helpful suggestion to include type assertions like this, which is cool, but we may not need it if we annotate the argument as `name: Option[str]` and then use MyPy to rule out type errors. Becoming MyPy-clean could be a heavy lift, however.

Reply:
Woof, I gave it a cursory shot and I'm not going to be able to do it in this PR! I don't understand the library well enough 😆